src/satosa/routing.py

```diff
@@ -155,13 +172,12 @@ def endpoint_routing(self, context):
         msg = "Routing path: {path}".format(path=context.path)
         logline = lu.LOG_FMT.format(id=lu.get_session_id(context.state), message=msg)
         logger.debug(logline)
-        path_split = context.path.split("/")
-        backend = path_split[0]
 
-        if backend in self.backends:
+        backend = self._find_backend(context.path)
```

Comment:
there was an idea here to reuse the method that is invoked to find a frontend;

see https://github.com/IdentityPython/SATOSA/pull/279

Reply:
I must say that I can't fully understand all the `_find_*` methods. My motives for yet another implementation is that it is looking at the very specific part of the request path and tries to find the backend **name**. I'm probably mistaken but I think that the reused version would try to match the backend **endpoint**, which is not present in a frontend request.